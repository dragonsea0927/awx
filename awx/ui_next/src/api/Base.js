import AxiosHTTP from './AxiosHTTP';

class Base {
  constructor(http = AxiosHTTP, baseURL) {
    this.http = http;
    this.baseUrl = baseURL;
  }

  create(data) {
    return this.http.post(this.baseUrl, data);
  }

  destroy(id) {
    return this.http.delete(`${this.baseUrl}${id}/`);
  }

  read(params) {
    return this.http.get(this.baseUrl, {
      params,
    });
  }

  readDetail(id) {
    return this.http.get(`${this.baseUrl}${id}/`);
  }

  readOptions() {
    return this.http.options(this.baseUrl);
  }

  replace(id, data) {
    return this.http.put(`${this.baseUrl}${id}/`, data);
  }

  update(id, data) {
    return this.http.patch(`${this.baseUrl}${id}/`, data);
  }

  copy(id, data) {
    return this.http.post(`${this.baseUrl}${id}/copy/`, data);
  }
}

export default Base;
