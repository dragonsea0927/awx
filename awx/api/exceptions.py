# Copyright (c) 2018 Ansible by Red Hat
# All Rights Reserved.

# Django
from django.utils.translation import ugettext_lazy as _

# Django REST Framework
from rest_framework.exceptions import ValidationError


class ActiveJobConflict(ValidationError):
    status_code = 409

    def __init__(self, active_jobs):
        super(ActiveJobConflict, self).__init__({
            "error": _("Resource is being used by running jobs."),
            "active_jobs": active_jobs
        })
