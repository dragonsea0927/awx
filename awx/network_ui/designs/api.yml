models:
- name: Device
  api: true
  v1_end_point: /network_ui/api/v1/device/
  v2_end_point: /api/v2/canvas/device/
  topology_id_query: topology_id
  v2_lookup_field: host_id
  create_transform:
      id: id
      name: name
      device_type: type
      x: x
      y: y
      interface_id_seq: interface_id_seq
      process_id_seq: process_id_seq
      host_id: host_id
      topology_id: topology_id
- name: Link
  api: true
  v1_end_point: /network_ui/api/v1/link/
  v2_end_point: /api/v2/canvas/link/
  topology_id_query: from_device__topology_id
  create_transform:
      id: id
      name: name
      from_device__id: from_device_id
      from_interface__id: from_interface_id
      to_device__id: to_device_id
      to_interface__id: to_interface_id
- name: Interface
  api: true
  v1_end_point: /network_ui/api/v1/interface/
  v2_end_point: /api/v2/canvas/interface/
  topology_id_query: device__topology_id
  create_transform:
      id: id
      name: name
      device__id: device_id
- name: Group
  api: true
  v1_end_point: /network_ui/api/v1/group/
  v2_end_point: /api/v2/canvas/group/
  topology_id_query: topology_id
- name: GroupDevice
  api: true
  v1_end_point: /network_ui/api/v1/groupdevice/
  v2_end_point: /api/v2/canvas/groupdevice/
  topology_id_query: group__topology_id
- name: Topology
  api: true
  v1_end_point: /network_ui/api/v1/topology/
  v2_end_point: /api/v2/canvas/topology/
  topology_id_query: topology_id
- name: TopologyInventory
  api: true
  v1_end_point: /network_ui/api/v1/topologyinventory/
  v2_end_point: /api/v2/canvas/topologyinventory/
  topology_id_query: topology_id
- name: Toolbox
  api: true
  v1_end_point: /network_ui/api/v1/toolbox/
  v2_end_point: /api/v2/canvas/toolbox/
- name: ToolboxItem
  api: true
  v1_end_point: /network_ui/api/v1/toolboxitem/
  v2_end_point: /api/v2/canvas/toolboxitem/
- name: Stream
  api: true
  v1_end_point: /network_ui/api/v1/stream/
  v2_end_point: /api/v2/canvas/stream/
  topology_id_query: from_device__topology_id
- name: Process
  api: true
  v1_end_point: /network_ui/api/v1/process/
  v2_end_point: /api/v2/canvas/process/
  topology_id_query: device__topology_id
