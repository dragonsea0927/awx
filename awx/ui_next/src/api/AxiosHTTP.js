import axios from 'axios';

import { encodeQueryString } from '../util/qs';

const AxiosHTTP = axios.create({
  xsrfCookieName: 'csrftoken',
  xsrfHeaderName: 'X-CSRFToken',
  paramsSerializer(params) {
    return encodeQueryString(params);
  },
});

export default AxiosHTTP;
