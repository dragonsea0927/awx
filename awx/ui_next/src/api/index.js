import Config from './models/Config';
import InstanceGroups from './models/InstanceGroups';
import JobTemplates from './models/JobTemplates';
import Jobs from './models/Jobs';
import Me from './models/Me';
import Organizations from './models/Organizations';
import Root from './models/Root';
import Teams from './models/Teams';
import UnifiedJobTemplates from './models/UnifiedJobTemplates';
import UnifiedJobs from './models/UnifiedJobs';
import Users from './models/Users';
import WorkflowJobTemplates from './models/WorkflowJobTemplates';

const ConfigAPI = new Config();
const InstanceGroupsAPI = new InstanceGroups();
const JobTemplatesAPI = new JobTemplates();
const JobsAPI = new Jobs();
const MeAPI = new Me();
const OrganizationsAPI = new Organizations();
const RootAPI = new Root();
const TeamsAPI = new Teams();
const UnifiedJobTemplatesAPI = new UnifiedJobTemplates();
const UnifiedJobsAPI = new UnifiedJobs();
const UsersAPI = new Users();
const WorkflowJobTemplatesAPI = new WorkflowJobTemplates();

export {
  ConfigAPI,
  InstanceGroupsAPI,
  JobTemplatesAPI,
  JobsAPI,
  MeAPI,
  OrganizationsAPI,
  RootAPI,
  TeamsAPI,
  UnifiedJobTemplatesAPI,
  UnifiedJobsAPI,
  UsersAPI,
  WorkflowJobTemplatesAPI,
};
