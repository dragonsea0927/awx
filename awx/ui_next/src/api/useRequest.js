import { useEffect, useState, useRef, useCallback } from 'react';

export default function useRequest(makeRequest, initialValue) {
  const [result, setResult] = useState(initialValue);
  const [error, setError] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  const isMounted = useRef(null);

  useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
    };
  }, []);

  return {
    result,
    error,
    isLoading,
    request: useCallback(async () => {
      setIsLoading(true);
      try {
        const response = await makeRequest();
        if (isMounted.current) {
          setResult(response);
        }
      } catch (err) {
        if (isMounted.current) {
          setError(err);
        }
      } finally {
        if (isMounted.current) {
          setIsLoading(false);
        }
      }
    }, [makeRequest]),
  };
}
