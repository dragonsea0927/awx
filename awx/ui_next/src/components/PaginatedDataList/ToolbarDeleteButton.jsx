import React, { useContext, useEffect, useState } from 'react';
import {
  func,
  bool,
  node,
  number,
  string,
  arrayOf,
  shape,
  checkPropTypes,
} from 'prop-types';
import styled from 'styled-components';
import {
  Alert,
  Badge,
  Button,
  DropdownItem,
  Tooltip,
} from '@patternfly/react-core';
import { withI18n } from '@lingui/react';
import { t } from '@lingui/macro';
import AlertModal from '../AlertModal';
import { KebabifiedContext } from '../../contexts/Kebabified';
import { getRelatedResourceDeleteCounts } from '../../util/getRelatedResourceDeleteDetails';

import ErrorDetail from '../ErrorDetail';

const WarningMessage = styled(Alert)`
  margin-top: 10px;
`;
const DetailsWrapper = styled.span`
  :not(:first-of-type) {
    padding-left: 10px;
  }
`;

const requiredField = props => {
  const { name, username, image } = props;
  if (!name && !username && !image) {
    return new Error(
      `One of 'name', 'username' or 'image' is required by ItemToDelete component.`
    );
  }
  if (name) {
    checkPropTypes(
      {
        name: string,
      },
      { name: props.name },
      'prop',
      'ItemToDelete'
    );
  }
  if (username) {
    checkPropTypes(
      {
        username: string,
      },
      { username: props.username },
      'prop',
      'ItemToDelete'
    );
  }
  if (image) {
    checkPropTypes(
      {
        image: string,
      },
      { image: props.image },
      'prop',
      'ItemToDelete'
    );
  }
  return null;
};

const ItemToDelete = shape({
  id: number.isRequired,
  name: requiredField,
  username: requiredField,
  image: requiredField,
  summary_fields: shape({
    user_capabilities: shape({
      delete: bool.isRequired,
    }).isRequired,
  }).isRequired,
});

function ToolbarDeleteButton({
  itemsToDelete,
  pluralizedItemName,
  errorMessage,
  onDelete,
  deleteDetailsRequests,
  warningMessage,
  deleteMessage,
  i18n,
  cannotDelete,
}) {
  const { isKebabified, onKebabModalChange } = useContext(KebabifiedContext);
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [deleteDetails, setDeleteDetails] = useState({});

  const deleteMessages = [warningMessage, deleteMessage].filter(
    message => message
  );

  const [deleteMessageError, setDeleteMessageError] = useState();
  const handleDelete = () => {
    onDelete();
    toggleModal();
  };

  const toggleModal = async isOpen => {
    if (itemsToDelete.length === 1 && deleteDetailsRequests?.length > 0) {
      const { results, error } = await getRelatedResourceDeleteCounts(
        deleteDetailsRequests
      );

      if (error) {
        setDeleteMessageError(error);
      } else {
        setDeleteDetails(results);
      }
    }

    setIsModalOpen(isOpen);
  };

  useEffect(() => {
    if (isKebabified) {
      onKebabModalChange(isModalOpen);
    }
  }, [isKebabified, isModalOpen, onKebabModalChange]);

  const renderTooltip = () => {
    const itemsUnableToDelete = itemsToDelete
      .filter(cannotDelete)
      .map(item => item.name || item.username)
      .join(', ');
    if (itemsToDelete.some(cannotDelete)) {
      return (
        <div>
          {errorMessage.length > 0
            ? `${errorMessage}: ${itemsUnableToDelete}`
            : i18n._(
                t`You do not have permission to delete ${pluralizedItemName}: ${itemsUnableToDelete}`
              )}
        </div>
      );
    }
    if (itemsToDelete.length) {
      return i18n._(t`Delete`);
    }
    return i18n._(t`Select a row to delete`);
  };

  const modalTitle = i18n._(t`Delete ${pluralizedItemName}?`);

  const isDisabled =
    itemsToDelete.length === 0 || itemsToDelete.some(cannotDelete);

  if (deleteMessageError) {
    return (
      <AlertModal
        isOpen={deleteMessageError}
        title={i18n._(t`Error!`)}
        onClose={() => {
          toggleModal(false);
          setDeleteMessageError();
        }}
      >
        <ErrorDetail error={deleteMessageError} />
      </AlertModal>
    );
  }

  return (
    <>
      {isKebabified ? (
        <Tooltip content={renderTooltip()} position="top">
          <DropdownItem
            key="add"
            isDisabled={isDisabled}
            component="button"
            onClick={() => toggleModal(true)}
          >
            {i18n._(t`Delete`)}
          </DropdownItem>
        </Tooltip>
      ) : (
        <Tooltip content={renderTooltip()} position="top">
          <div>
            <Button
              variant="secondary"
              aria-label={i18n._(t`Delete`)}
              onClick={() => toggleModal(true)}
              isAriaDisabled={isDisabled}
            >
              {i18n._(t`Delete`)}
            </Button>
          </div>
        </Tooltip>
      )}

      {isModalOpen && !deleteMessageError && (
        <AlertModal
          variant="danger"
          title={modalTitle}
          isOpen={isModalOpen}
          onClose={() => toggleModal(false)}
          actions={[
            <Button
              key="delete"
              variant="danger"
              aria-label={i18n._(t`confirm delete`)}
              onClick={handleDelete}
            >
              {i18n._(t`Delete`)}
            </Button>,
            <Button
              key="cancel"
              variant="link"
              aria-label={i18n._(t`cancel delete`)}
              onClick={() => toggleModal(false)}
            >
              {i18n._(t`Cancel`)}
            </Button>,
          ]}
        >
          <div>{i18n._(t`This action will delete the following:`)}</div>
          {itemsToDelete.map(item => (
            <span key={item.id}>
              <strong>{item.name || item.username || item.image}</strong>
              <br />
            </span>
          ))}
          {itemsToDelete.length === 1 &&
            Object.values(deleteDetails).length > 0 && (
              <WarningMessage
                variant="warning"
                isInline
                title={
                  <div>
                    {deleteMessages.map(message => (
                      <div aria-label={message} key={message}>
                        {message}
                      </div>
                    ))}
                    {itemsToDelete.length === 1 && (
                      <>
                        <br />
                        {Object.entries(deleteDetails).map(([key, value]) => (
                          <DetailsWrapper
                            key={key}
                            aria-label={`${key}: ${value}`}
                          >
                            <span>{key}</span> <Badge>{value}</Badge>
                          </DetailsWrapper>
                        ))}
                      </>
                    )}
                  </div>
                }
              />
            )}
          {itemsToDelete.length > 1 && (
            <WarningMessage
              variant="warning"
              isInline
              title={deleteMessages.map(message => (
                <div>{message}</div>
              ))}
            />
          )}
        </AlertModal>
      )}
    </>
  );
}

ToolbarDeleteButton.propTypes = {
  onDelete: func.isRequired,
  itemsToDelete: arrayOf(ItemToDelete).isRequired,
  pluralizedItemName: string,
  errorMessage: string,
  warningMessage: node,
  cannotDelete: func,
};

ToolbarDeleteButton.defaultProps = {
  pluralizedItemName: 'Items',
  errorMessage: '',
  warningMessage: null,
  cannotDelete: item => !item.summary_fields.user_capabilities.delete,
};

export default withI18n()(ToolbarDeleteButton);
