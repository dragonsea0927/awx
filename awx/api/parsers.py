# Python
from collections import OrderedDict
import json

# Django
from django.conf import settings
from django.utils import six
from django.utils.translation import ugettext_lazy as _

# Django REST Framework
from rest_framework import parsers
from rest_framework.exceptions import ParseError


class JSONParser(parsers.JSONParser):
    """
    Parses JSON-serialized data, preserving order of dictionary keys.
    """

    def parse(self, stream, media_type=None, parser_context=None):
        """
        Parses the incoming bytestream as JSON and returns the resulting data.
        """
        parser_context = parser_context or {}
        encoding = parser_context.get('encoding', settings.DEFAULT_CHARSET)

        try:
            data = stream.read().decode(encoding)
            if not data:
                return {}
            obj = json.loads(data, object_pairs_hook=OrderedDict)
            if not isinstance(obj, dict) and obj is not None:
                raise ParseError(_('JSON parse error - not a JSON object'))
            return obj
        except ValueError as exc:
            raise ParseError(_('JSON parse error - %s\nPossible cause: trailing comma.' % six.text_type(exc)))
