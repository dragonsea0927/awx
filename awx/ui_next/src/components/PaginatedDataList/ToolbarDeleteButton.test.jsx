import React from 'react';
import { act } from 'react-dom/test-utils';
import {
  mountWithContexts,
  waitForElement,
} from '../../../testUtils/enzymeHelpers';
import { CredentialsAPI } from '../../api';
import ToolbarDeleteButton from './ToolbarDeleteButton';

jest.mock('../../api');

const itemA = {
  id: 1,
  name: 'Foo',
  summary_fields: { user_capabilities: { delete: true } },
};
const itemB = {
  id: 1,
  name: 'Foo',
  summary_fields: { user_capabilities: { delete: false } },
};
const itemC = {
  id: 1,
  username: 'Foo',
  summary_fields: { user_capabilities: { delete: false } },
};

describe('<ToolbarDeleteButton />', () => {
  let deleteDetailsRequests;
  beforeEach(() => {
    deleteDetailsRequests = [
      {
        label: 'Workflow Job Template Node',
        request: CredentialsAPI.read.mockResolvedValue({ data: { count: 1 } }),
      },
    ];
  });
  test('should render button', () => {
    const wrapper = mountWithContexts(
      <ToolbarDeleteButton onDelete={() => {}} itemsToDelete={[]} />
    );
    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper.find('ToolbarDeleteButton')).toMatchSnapshot();
  });

  test('should open confirmation modal', async () => {
    let wrapper;
    await act(async () => {
      wrapper = mountWithContexts(
        <ToolbarDeleteButton
          onDelete={() => {}}
          itemsToDelete={[itemA]}
          deleteDetailsRequests={deleteDetailsRequests}
          deleteMessage="Delete this?"
          warningMessage="Are you sure to want to delete this"
        />
      );
    });

    expect(wrapper.find('Modal')).toHaveLength(0);
    await act(async () => {
      wrapper.find('button').prop('onClick')();
    });
    await waitForElement(wrapper, 'Modal', el => el.length > 0);
    expect(CredentialsAPI.read).toBeCalled();
    expect(wrapper.find('Modal')).toHaveLength(1);
    expect(
      wrapper.find('span[aria-label="Workflow Job Template Node: 1"]')
    ).toHaveLength(1);
    expect(wrapper.find('div[aria-label="Delete this?"]')).toHaveLength(1);
  });

  test('should open delete error modal', async () => {
    const request = [
      {
        label: 'Workflow Job Template Node',
        request: CredentialsAPI.read.mockRejectedValue(
          new Error({
            response: {
              config: {
                method: 'get',
                url: '/api/v2/credentals',
              },
              data: 'An error occurred',
              status: 403,
            },
          })
        ),
      },
    ];
    let wrapper;
    await act(async () => {
      wrapper = mountWithContexts(
        <ToolbarDeleteButton
          onDelete={() => {}}
          itemsToDelete={[itemA]}
          deleteDetailsRequests={request}
          deleteMessage="Delete this?"
          warningMessage="Are you sure to want to delete this"
        />
      );
    });

    expect(wrapper.find('Modal')).toHaveLength(0);
    await act(async () => wrapper.find('button').simulate('click'));
    await waitForElement(wrapper, 'Modal', el => el.length > 0);
    expect(CredentialsAPI.read).toBeCalled();

    wrapper.update();

    expect(wrapper.find('AlertModal[title="Error!"]')).toHaveLength(1);
  });

  test('should invoke onDelete prop', () => {
    const onDelete = jest.fn();
    const wrapper = mountWithContexts(
      <ToolbarDeleteButton onDelete={onDelete} itemsToDelete={[itemA]} />
    );
    wrapper.find('button').simulate('click');
    wrapper.update();
    wrapper
      .find('ModalBoxFooter button[aria-label="confirm delete"]')
      .simulate('click');
    wrapper.update();
    expect(onDelete).toHaveBeenCalled();
    expect(wrapper.find('Modal')).toHaveLength(0);
  });

  test('should disable button when no delete permissions', () => {
    const wrapper = mountWithContexts(
      <ToolbarDeleteButton onDelete={() => {}} itemsToDelete={[itemB]} />
    );
    expect(wrapper.find('button[disabled]')).toHaveLength(1);
  });

  test('should render tooltip', () => {
    const wrapper = mountWithContexts(
      <ToolbarDeleteButton onDelete={() => {}} itemsToDelete={[itemA]} />
    );
    expect(wrapper.find('Tooltip')).toHaveLength(1);
    expect(wrapper.find('Tooltip').prop('content')).toEqual('Delete');
  });

  test('should render tooltip for username', () => {
    const wrapper = mountWithContexts(
      <ToolbarDeleteButton onDelete={() => {}} itemsToDelete={[itemC]} />
    );
    expect(wrapper.find('Tooltip')).toHaveLength(1);
    expect(wrapper.find('Tooltip').prop('content').props.children).toEqual(
      'You do not have permission to delete Items: Foo'
    );
  });
});
