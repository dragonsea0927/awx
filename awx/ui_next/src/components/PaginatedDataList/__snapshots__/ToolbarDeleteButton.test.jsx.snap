// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolbarDeleteButton /> should render button 1`] = `
<ToolbarDeleteButton
  i18n={"/i18n/"}
  itemsToDelete={Array []}
  onDelete={[Function]}
  pluralizedItemName="Items"
>
  <Tooltip
    appendTo={[Function]}
    aria="describedby"
    boundary="window"
    className=""
    content="Select a row to delete"
    distance={15}
    enableFlip={true}
    entryDelay={500}
    exitDelay={500}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    id=""
    isAppLauncher={false}
    isContentLeftAligned={false}
    isVisible={false}
    maxWidth="18.75rem"
    position="top"
    tippyProps={Object {}}
    trigger="mouseenter focus"
    zIndex={9999}
  >
    <PopoverBase
      appendTo={[Function]}
      aria="describedby"
      arrow={true}
      boundary="window"
      content={
        <div
          className=""
          id=""
          role="tooltip"
        >
          <TooltipContent
            isLeftAligned={false}
          >
            Select a row to delete
          </TooltipContent>
        </div>
      }
      delay={
        Array [
          500,
          500,
        ]
      }
      distance={15}
      flip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      lazy={true}
      maxWidth="18.75rem"
      onCreate={[Function]}
      placement="top"
      popperOptions={
        Object {
          "modifiers": Object {
            "hide": Object {
              "enabled": true,
            },
            "preventOverflow": Object {
              "enabled": true,
            },
          },
        }
      }
      theme="pf-tooltip"
      trigger="mouseenter focus"
      zIndex={9999}
    >
      <div>
        <Component
          aria-label="Delete"
          isDisabled={true}
          onClick={[Function]}
          variant="danger"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "aria-label": "Delete",
                "children": "Delete",
                "isDisabled": true,
                "onClick": [Function],
                "variant": "danger",
              }
            }
            consumerContext={null}
          >
            <Button
              aria-label="Delete"
              isDisabled={true}
              onClick={[Function]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              variant="danger"
            >
              <button
                aria-disabled={null}
                aria-label="Delete"
                className="pf-c-button pf-m-danger"
                disabled={true}
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                Delete
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
      </div>
      <Portal
        containerInfo={
          <div>
            <div
              class=""
              id=""
              role="tooltip"
            >
              <div
                class="pf-c-tooltip__content"
              >
                Select a row to delete
              </div>
            </div>
          </div>
        }
      >
        <div
          className=""
          id=""
          role="tooltip"
        >
          <TooltipContent
            isLeftAligned={false}
          >
            <div
              className="pf-c-tooltip__content"
            >
              Select a row to delete
            </div>
          </TooltipContent>
        </div>
      </Portal>
    </PopoverBase>
  </Tooltip>
</ToolbarDeleteButton>
`;
